lib/event.ts

```diff
@@ -0,0 +1,57 @@
+import { Config } from './config';
+
+export type EventHeaders = {
+	appName: string;
+	resource: string;
+	origin: string;
+	action: string;
+	destination: string;
+	schemaVersion: number;
+};
+
+export class Event {
+	private headers: EventHeaders;
+	constructor(
+		private name: string,
+		private body: Record<string, any>,
+		private schemaVersion = 1.0,
+	) {}
+
+	payload() {
+		return {
+			headers: this.headers,
+			body: this.body,
+		};
+	}
+
+	hasBody() {
+		return this.body !== undefined;
+	}
+
+	hasName() {
+		return this.name !== undefined;
+	}
+
+	private buildPayload() {
+		//nao faz o menor sentido
```

Comment:
Why doesn't it make any sense? :thinking:

Reply:
becuase there is a property "body" inside a "body" object, it seems a bit confuse, but honestly i didn't mean to commit with this comment hahahaaha